tests/visualization_tests/matplotlib_tests/test_pareto_front.py

```diff
@@ -79,23 +79,25 @@ def test_plot_pareto_front_2d(
     plt.savefig(BytesIO())
 
     # Test with `target_names` argument.
-    with pytest.raises(ValueError):
+    error_message = "The length of `target_names` is supposed to be 2."
```

Comment:
I think tests should be passed even if error messages are changed. Error messages are just implementation details and it's out of specification of `plot_pareto_front()`. To reduce the maintainability of these tests, How about reverting these changes?

Reply:
I think there is value in maintaining the error messages that are displayed to users, as I have had users complain in the past when error messages were unexpectedly changed to something that was difficult to understand.

Reply:
Thank you for your discussion! I also cared about maintainability because this plot function has eight `ValueError` sources, so I found difficult it to which test case corresponds to which `ValueError` in `optuna/visualization/_pareto_front.py`.

Reply:
Thank you for your comments!

> as I have had users complain in the past when error messages were unexpectedly changed to something that was difficult to understand.

Hmm... In this case, the problem is that error messages are changed to something that was difficult to understand. It sounds that the change of error messages itself was not a problem.

> I found difficult it to which test case corresponds to which ValueError in optuna/visualization/_pareto_front.py. This is why I've (partially) added error message checks.

Thank you for sharing your opinion! I think we don't need to check whether each test case corresponds to which ValueError. It depends too much on the implementation details.

But anyway, as @HideakiImamura said, it might be a bit controversial. We should not deepen this discussion too much. I will approve this PR even if these changes won't be reverted :bow: